### DISCUSSION TITLE
Menubar Component Sample Code Navigation Issue

### DISCUSSION THREAD

**Discussion by RickKukiela** [primefaces/community#222] [PrimeVue]
I was recently implementing a `<Menubar>` component (first time using PrimeAnything).

To get started, I ended up copying and pasting the sample code (from the PrimeFaces.org docs page) that allows customization of the `#item` slot:

```vue:
<Menubar :model="items">
    <template #item="{ label, item, props, root, hasSubmenu }">
        <router-link v-if="item.route" v-slot="routerProps" :to="item.route" custom>
            <a :href="routerProps.href" v-bind="props.action">
                <span v-bind="props.icon" />
                <span v-bind="props.label">{{ label }}</span>
            </a>
        </router-link>
        <a v-else :href="item.url" :target="item.target" v-bind="props.action">
            <span v-bind="props.icon" />
            <span v-bind="props.label">{{ label }}</span>
            <span :class="[hasSubmenu && (root ? 'pi pi-fw pi-angle-down' : 'pi pi-fw pi-angle-right')]" v-bind="props.submenuicon" />
        </a>
    </template>
</Menubar>
```

The problem with the code is that it's missing the `@click="routerProps.navigate"` portion on the `<a>` tag inside the `<router-link>` custom slot.

I would imagine that /most/ users would want the links in the menu to route directly to the linked VUE "page" and not cause a complete "reboot" of a SPA [app] interface, so I recommend that this be added to the docs sample code for the next person. If its preferred to leave it vanilla like this, then I do recommend adding a note about this on the page. 

I'm also quite curious as to why `v-bind="props.action"` does not set this `@click` action by default. Granted I do not fully understand how that works exactly and if its something the `<Menubar>` component `#item` slot can provide or if that is a feature that the folks over @ VueRouter would have explicit control over.

**Comment by tugcekucukoglu** [comment]
In the Menubar demos, the example code is not written for the navigation because there is no navigation example. For instance, if you look at the TabMenu examples, you can find the relevant structure for the click event.


```
<TabMenu v-model:activeIndex="active" :model="items">
    <template #item="{ label, item, props }">
        <router-link v-if="item.route" v-slot="routerProps" :to="item.route" custom>
            <a :href="routerProps.href" v-bind="props.action" @click="($event) => routerProps.navigate($event)" @keydown.enter.space="($event) => routerProps.navigate($event)">
                <span v-bind="props.icon" />
                <span v-bind="props.label">{{ label }}</span>
            </a>
        </router-link>
        <a v-else :href="item.url" :target="item.target" v-bind="props.action">
            <span v-bind="props.icon" />
            <span v-bind="props.label">{{ label }}</span>
        </a>
    </template>
</TabMenu>
```

**Reply by RickKukiela** [reply]
[tugcekucukoglu](https://github.com/tugcekucukoglu) Thank you for the reply.

I could argue that the `<Menubar>` examples contain `Users > New` `Upload` and `Quit`; all of which _could be_ "navigation routes" depending on the app and implementation. I'm not sure why we would make the assumption that no one using this this component would expect to have links in it for the purpose of navigating their app. I'm one such person that only intended to ever use this component in this manner. While it was a good learning experience on the internal workings of `vue-router` that I'm surely better off knowing now, it would have been nice to not have to learn about this "the hard way".

While I am glad that there is an example of properly binding the `@click` property of the nested `<a>` tag on the `<TabMenu>` component, that did not help me at all since I was not using that component and never even looked at it.

I stand by my recommendation that for the benefit of the reader and maximum clarity, the `<Menubar>` component docs should have it's examples updated to apply the click handler, possibly via an additional `v-if-else` based on an extra possible data point on the item props like `internal: boolean` or something, so the user can see the difference on how an internal vs external route is handled in combination with the `custom` property on the `vue-router` usage.
